### DISCUSSION TITLE
tasmota 13.0.0.3 discovery doesn't contain sonoff SNZB-02D temperature sensors

### DISCUSSION THREAD

**Discussion by zerograv90** [arendst/Tasmota#19201] [Q&A] [answered]
Hi, i have installed tasmota on an sonoff zbbridge-p (pro) und coupled some sonoff SNZB-02D sensors.
mqtt with an mosquitto broker is configured.

Here are some infos to the installed tasmota version:
![image](https://github.com/arendst/Tasmota/assets/130251252/0037f917-030e-4c64-a9e3-4a6e762e70b6)

The devices showing correctly in the main menu
![image](https://github.com/arendst/Tasmota/assets/130251252/2ed130ad-d58c-4fda-8c27-58fc453f2a5a)

Sensor if delivering a valid value in the tele/tasmota../sensor topic:
![image](https://github.com/arendst/Tasmota/assets/130251252/af5dc83e-a760-4bb4-827e-8b6ae34f97aa)

But the sensors doesn't show up in the tasmota/discovery channel:
![image](https://github.com/arendst/Tasmota/assets/130251252/8228761d-4041-49e6-9059-9d11178782bf)

I am using the tasmota intregation in Homeassistant. So no entitys are listed currently for the sensors.

Am i missing something? Should the connected sensors show up in in the autodiscovery?

**Comment by sfromis** [comment, marked as answer]
That sensor payload you see there is a special one from Zigbee, not the general one used for discovery. Thus, there is no auto discovery of Zigbee devices, and at boot time the Zigbee sensors would not have been seen anyway.

BTW, I'd recommend options `ZbNoPrefix/SetOption100` and `ZbNameKey/SetOption83` to have the Zigbee sensor payload become more similar to the regular sensor payload, even if this still does not make it part of the discovery payload at boot time.

Since you are using a ESP32-based coordinator device, you may find the project [Home Assistant Controls for Tasmota](https://github.com/fmtr/hct) useful for easier Home Assistant integration. Not using HA, I have no experience with this library, but the description does mention sensors.

**Comment by zerograv90** [comment]
Thank you very much for the explanation. 
The options made the published topic a bit more readable 😊

Sadly the Home assistant controls for tasmota doesn't work eg. the installation fails on the zbbridge-p

Currently i am testing the zha / TCP ZBBridge Pro config. 

I think if this is unstable or unreliable i will add manual the mqtt entity to home asisstant.

**Reply by Spilho** [reply]
Have you solved this issue?
I am having same issue and using same setup. 
Any suggestion or advise would be helpful.

**Reply by sfromis** [reply]
There is no issue here, it is just how it works with Zigbee devices. Tasmota cannot predict what might be relevant to include in the discovery payloads. Hence manual HA config is a reasonable way. Not sure if the HCT project I mentioned would helpful for this.

**Reply by Spilho** [reply]
HCT project was not helpful or it was just my lack of knowledge.
Going to google how to configure it manually in HA.
Thanks for fast respond.

**Reply by sfromis** [reply]
BTW. If you are intending to use the Zigbee data **exclusively** with HA, it could also be a reasonable approach to use ZHA, where Tasmota gives up any control over Zigbee, to let ZHA fully take over the coordinator role.

**Comment by zerograv90** [comment]
ZHA was the way which worked for me with HA.
But i have only HA which consumes my sensor data.

sfromis ***@***.***> schrieb am Di., 26. Dez. 2023, 23:06:
